### PR TITLE
Make mice great again

### PR DESCRIPTION
**Pull request by BadRyuner**
## About the PR

Added the ability for mice to disarm mousetraps. 

## Why / Balance
The chances of disarming may not be very balanced, but it's better than nothing. 

Players surround the entire kitchen with mousetraps. This completely kills the gameplay for the mouse. This PR gives a small chance for mouse players to sneak in there and bring CHAOS.

## Technical details
Added Verb for the mousetrap. When activated it randomly either kills the mouse, disarms the mousetrap and damages the mouse (a quarter of the mousetrap damage), or just disarms the mousetrap.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

[example.webm](https://github.com/space-wizards/space-station-14/assets/54708336/c744633b-364e-4711-b901-56b1558b9c84)

## Breaking changes

**Changelog**

- add: Mice can disarm a mousetrap

### CODE REVIEW

**Comment by deltanedas** [comment]
mousetraps in the video look disarmed already...

**Comment by UbaserB** [comment]
great idea

**Comment by Partmedia** [comment]
> Players surround the entire kitchen with mousetraps. This completely kills the gameplay for the mouse.

I don't think so; they can go find food elsewhere. The only reason players surround the entire kitchen with mousetraps is if mice are being excessively egregious.

**Comment by deltanedas** [comment]
yeah i very rarely see chefs mine the kitchen, severe lack of pros

**Comment by forgotmyotheraccount** [comment]
The high rate of failure seems fair, I don't see an issue with it. Maybe add a doafter to prevent lucky players from blitzing through a mined kitchen in seconds.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Mousetrap/MousetrapSystem.cs [hunk 1]
```diff
@@ -61,6 +69,65 @@ private void OnTrigger(EntityUid uid, MousetrapComponent component, TriggerEvent
...
+            {
+                if (!component.IsActive)
+                {
+                    _popupSystem.PopupEntity(Loc.GetString("mousetrap-is-not-activated"), uid, args.User);
+                    return;
+                }
+
+                _audio.PlayPvs(Comp<EmitSoundOnTriggerComponent>(uid).Sound, uid); // play snAp
+
+                if (_random.Prob(MathF.Min(0.99f, 0.15f * component.Difficulty))) // Chance to die
```
Comment (metalgearsloth) [inline Content.Server/Mousetrap/MousetrapSystem.cs:93]: Don't use a dummy min statement, just check the value.

Also having a random chance to die makes this feature nearly pointless when they can just go around the mousetrap 99% of the time.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Forgot to mention but this should also be added to shared and predicted.

**Pull request closed without merging**